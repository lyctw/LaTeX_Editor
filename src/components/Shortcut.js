import React from 'react';
// import add_svg from '../img/set1/add.svg'
// import minus_svg from '../img/set1/minus.svg'
import times_svg from '../img/set1/times.svg'
import div_svg from '../img/set1/div.svg'  
import amalg_svg from '../img/set1/amalg.svg'
import star_svg from '../img/set1/star.svg'
import pm_svg from '../img/set1/pm.svg'

const Shortcut = ({handleAppendTex}) => {
    return (
        <>
            <details><summary>Shortcut</summary>

                {/* <button onClick={() => {handleAppendTex(' + ')}}><img src={add_svg} /></button> */}
                {/* <button onClick={() => {handleAppendTex(' - ')}}><img src={minus_svg} /></button> */}
                <button onClick={() => {handleAppendTex(String.raw` \times `)}}><img src={times_svg} /></button>
                <button onClick={() => {handleAppendTex(String.raw` \div `)}}><img src={div_svg} /></button>
                <button onClick={() => {handleAppendTex(String.raw` \amalg `)}}><img src={amalg_svg} /></button>
                <button onClick={() => {handleAppendTex(String.raw` \star `)}}><img src={star_svg} /></button>
                <button onClick={() => {handleAppendTex(String.raw` \pm `)}}><img src={pm_svg} /></button>
                
            
            </details>
        </>
    );
}

export default Shortcut;