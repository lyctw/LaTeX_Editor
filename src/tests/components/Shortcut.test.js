import React from 'react'
import { shallow } from 'enzyme'
import Shortcut from '../../components/Shortcut'
import toJson from 'enzyme-to-json';
 
test('should render Shortcut', () => {
    const wrapper = shallow(<Shortcut />);
    expect(toJson(wrapper)).toMatchSnapshot();
});
