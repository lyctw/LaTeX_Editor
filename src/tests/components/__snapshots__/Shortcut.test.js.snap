// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Shortcut 1`] = `
<div>
  <details>
    <summary>
      Shortcut
    </summary>
    <button
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
    </button>
  </details>
</div>
`;
